app/src/modules/analytics/events/common/pricing/index.ts

```diff
@@ -1,9 +1,25 @@
 import { trackEvent } from "modules/analytics";
 import { PRICING } from "../constants";
+import { FeatureLimitType } from "hooks/featureLimiter/types";
+import { RuleType } from "types";
 
 type ViewPricingPlansClickSource = "crown" | "my_profile" | "feature_limiter_banner" | "workspace_upgrade";
 
 export const trackViewPricingPlansClicked = (source: ViewPricingPlansClickSource) => {
   const params = { source };
   trackEvent(PRICING.VIEW_PRICING_PLANS_CLICKED, params);
 };
+
+export type FeatureType = Lowercase<keyof typeof FeatureLimitType>;
+
+export type PaidFeatureNudgeViewedSource =
+  | "rule_dropdown"
+  | "rule_selection_screen"
+  | "share_button"
+  | "share_modal"
+  | Lowercase<keyof typeof RuleType>;
+
```

Comment:
Shouldn't we use enum here? One benefit I can see is that while assingment we can use `PaidFeatureNudgeViewedSource.rule_dropdown`. WDYT?

Reply:
For this we have to import the enum `PaidFeatureNudgeViewedSource` everywhere were we are showing premium icon which I feel was not needed.